### PR TITLE
Integrate with AUHMetadataPB from pg -> yb_client -> docdb

### PR DESCRIPTION
**Pull request by amitanandaiyer**
- Integrate with ASaha's AUHMetadataPB
 - trying to add more fields to AUHMetadata
 - populate client_node_id based on Connection::Endpoint/remote_address()
 - Include AUHMetadata in all the yb/async_rpc-s
 - use AUHMetadata in pg_session

### CODE REVIEW

File: src/yb/rpc/yb_rpc.cc [hunk 1]
```diff
@@ -320,7 +329,12 @@ bool YBInboundCall::DumpPB(const DumpRunningRpcsRequestPB& req,
     resp->set_trace_buffer(my_trace->DumpToString(true));
   }
   if (req.get_wait_state()) {
-    resp->set_wait_state(wait_state()->ToString());
+    auto wait_state = this->wait_state();
+    if (wait_state) {
+      resp->set_wait_state(wait_state->ToString());
```
Comment (abhinab-yb) [inline src/yb/rpc/yb_rpc.cc:334]: Should the wait state be serialized to a PB instead of string? That way it would be easier in pggate side to parse it

Comment (amitanandaiyer) [reply src/yb/rpc/yb_rpc.cc]: I have it in the next part. 

updated here.

File: src/yb/util/wait_state.h [hunk 2]
```diff
@@ -81,11 +81,59 @@ YB_DEFINE_ENUM(
     (GetSafeTime)(GetSubDoc))
 
 struct AUHMetadata {
-  std::string request_id;
-  // TBD: other fields as required.
+  std::string top_level_request_id;
+  std::string top_level_node_id;
+  int64_t query_id = 0;
+  int64_t current_request_id = 0;
```
[the remark this answers is not in the document]
Comment (amitanandaiyer) [reply src/yb/util/wait_state.h]: good point. thanks!!

**Comment by amitanandaiyer** [comment]
Discarded in favor of https://github.com/yugabyte/yugabyte-db/pull/18129

**Pull request closed without merging**